### PR TITLE
refactor: extract mun_hir_input

### PR DESCRIPTION
**Pull request by baszalmstra**
Extracts the input part of the hir database into its own seperate crate: `mun_hir_input`.

The goals is to create a cleaner API separation and at the same time decrease incremental compilation time by splitting crates.

### CODE REVIEW

File: crates/mun_hir_input/src/package_set.rs [hunk 1]
```diff
@@ -4,26 +4,38 @@ use rustc_hash::FxHashMap;
...
 pub struct PackageSet {
     arena: FxHashMap<PackageId, PackageData>,
 }
 
 impl PackageSet {
-    /// Adds a new package to the package set
+    /// Returns an empty package set
+    pub fn new() -> Self {
+        Self::default()
+    }
```
Comment (Wodann) [inline crates/mun_hir_input/src/package_set.rs:35]: Is this necessary?

Comment (baszalmstra) [inline crates/mun_hir_input/src/package_set.rs:35]:
```suggestion
```

File: crates/mun_hir_input/src/snapshots/mun_hir_input__module_tree__test__module_tree.snap [hunk 2]
```diff
@@ -0,0 +1,97 @@
+---
+source: crates/mun_hir_input/src/module_tree.rs
```
Comment (Wodann) [inline crates/mun_hir_input/src/snapshots/mun_hir_input__module_tree__test__module_tree.snap:2]: Did you delete the snapshot from the `mun_hir` directory?

**Review by Wodann — APPROVED** [review]
Two comments but feel free to address them and merge yourself.

**Pull request merged**